src/target/riscv/riscv-011.c

```diff
@@ -730,10 +725,17 @@ static int wait_for_debugint_clear(struct target *target, bool ignore_first)
 		 * result of the read that happened just before debugint was set.
 		 * (Assuming the last scan before calling this function was one that
 		 * sets debugint.) */
-		read_bits(target);
+		read_bits(target, NULL);
 	}
 	while (1) {
-		bits_t bits = read_bits(target);
+		bits_t bits = {
+			.haltnot = 0,
+			.interrupt = 0
+		};
+		if (read_bits(target, &bits) != ERROR_OK) {
+			LOG_ERROR("read_bits fails");
```

Comment:
We don't need this generic error message, since read_bits() already provides a more specific error message.

Reply:
I've decided to add those for consistency. Meaning, if we decide to refactor/change this code in the future, we can have a higher chance that at least some error message is printed.  Excessive printing should not be a problem - since these kinds of messages indicate that something is terribly wrong anyway. I don't insist however, so if you think that it's better to remove them, I'll do it. Nevertheless, could you revise your comments?

Reply:
What are you being consistent with? Generally the approach is to print one error as specific as possible, ideally in the place that the error is first generated. Then the other layers can simply propagate the error up. There are exceptions where the caller has more information to add, but in this case the caller error message doesn't add any extra information.

Reply:
I just want to add more information in log (for example, we can understand what `read_bits` was called and failed in this function because of this error message), but if it's not necessary, I'll remove it

Reply:
Done